apps/design/frontend/src/election_info_screen.tsx

```diff
@@ -1,95 +1,106 @@
 import React, { useState } from 'react';
-import {
-  Election,
-  ElectionId,
-  ElectionIdSchema,
-  unsafeParse,
-} from '@votingworks/types';
+import { ElectionIdSchema, unsafeParse } from '@votingworks/types';
 import {
   Button,
   H1,
   MainContent,
   MainHeader,
   SegmentedButton,
 } from '@votingworks/ui';
+import type { ElectionInfo } from '@votingworks/design-backend';
 import { useHistory, useParams } from 'react-router-dom';
 import styled from 'styled-components';
 import { z } from 'zod';
 import { DateWithoutTime } from '@votingworks/basics';
 import {
   deleteElection,
   getBallotsFinalizedAt,
-  getElection,
-  updateElection,
+  getElectionInfo,
+  updateElectionInfo,
 } from './api';
 import { FieldName, Form, FormActionsRow, InputGroup } from './layout';
 import { ElectionNavScreen } from './nav_screen';
 import { routes } from './routes';
 import { ImageInput } from './image_input';
 
-type ElectionInfo = Pick<
-  Election,
-  'title' | 'date' | 'type' | 'state' | 'county' | 'seal'
->;
-
 const SealImageInput = styled(ImageInput)`
   img {
     width: 10rem;
   }
 `;
 
-function hasBlankElectionInfo(election: Election): boolean {
+function hasBlankElectionInfo(electionInfo: ElectionInfo): boolean {
   return (
-    election.title === '' &&
-    election.state === '' &&
-    election.county.name === '' &&
-    election.seal === ''
+    !electionInfo.title &&
+    !electionInfo.state &&
+    !electionInfo.jurisdiction &&
+    !electionInfo.seal
   );
 }
 
 function ElectionInfoForm({
-  electionId,
-  savedElection,
+  savedElectionInfo,
   ballotsFinalizedAt,
 }: {
-  electionId: ElectionId;
-  savedElection: Election;
+  savedElectionInfo: ElectionInfo;
   ballotsFinalizedAt: Date | null;
 }): JSX.Element {
   const [isEditing, setIsEditing] = useState(
     // Default to editing for newly created elections
-    hasBlankElectionInfo(savedElection)
+    hasBlankElectionInfo(savedElectionInfo)
   );
-  const [electionInfo, setElectionInfo] = useState<ElectionInfo>(savedElection);
-  const updateElectionMutation = updateElection.useMutation();
+  const [editedElectionInfo, setEditedElectionInfo] =
+    useState(savedElectionInfo);
+  const updateElectionInfoMutation = updateElectionInfo.useMutation();
   const deleteElectionMutation = deleteElection.useMutation();
   const history = useHistory();
+  const electionInfo = isEditing
+    ? editedElectionInfo ?? savedElectionInfo
+    : savedElectionInfo;
+
+  function resetForm() {
+    setIsEditing(false);
+    setEditedElectionInfo(savedElectionInfo);
+  }
 
   function onSubmit() {
-    updateElectionMutation.mutate(
-      { electionId, election: { ...savedElection, ...electionInfo } },
-      { onSuccess: () => setIsEditing(false) }
-    );
+    updateElectionInfoMutation.mutate(electionInfo, {
+      onSuccess: resetForm,
```

Comment:
I'm a little worried that calling `resetForm` here will call `setEditedElectionInfo(savedElectionInfo);` with the old `savedElectionInfo` and overwrite the changes the user made. Maybe that's not actually happening, but seems safer to not do this and revert to the logic we had before. Unless I'm misreading

Reply:
[ad7f4c3](https://github.com/votingworks/vxsuite/pull/5915/commits/ad7f4c34da68ccf1d4299e342afc47c8a34958fb)